src/jump.jl

```diff
@@ -681,6 +687,38 @@ function pass_primal_info(single_blm, primal, info::BilevelVariableInfo)
     return
 end
 
+function pass_necessary_dual_info(
+    single_blm,
+    info,
+    consider_constrained_variables::Bool,
+    lower_primal_dual_map,
+    lower_dual_to_sblm,
+    ctr_idx::MOI.ConstraintIndex{F,S},
+) where {F<:Union{MOI.VariableIndex,MOI.VectorOfVariables},S}
+    if consider_constrained_variables
+        return
+    else
+        pre_duals = lower_primal_dual_map.primal_con_dual_var[ctr_idx] # vector
+        duals = map(x -> lower_dual_to_sblm[x], pre_duals)
+        pass_dual_info(single_blm, duals, info)
+    end
+    return
+end
+
+function pass_necessary_dual_info(
```

Comment:
same function twice?

Reply:
sorry, thought I already replied to these... 
now a conditional